meson.build

```diff
@@ -3318,6 +3318,14 @@ custom_target(
                    '} >@OUTPUT@'],
         build_by_default : true)
 
+# We intentionally do not inline initializations with definitions for
+# a bunch of _cleanup_ variables in tests, to ensure valgrind is triggered.
+# This triggers a lot of false maybe-uninitialized false positives. Suppress them.
+if cc.has_argument('-Wno-maybe-uninitialized')
+        no_uninit = ['-Wno-maybe-uninitialized']
+else
+        no_uninit = []
```

Comment:
Maybe
```python
no_uninit = cc.first_supported_argument('-Wno-maybe-uninitialized')
```

(Untested, but should work, see https://mesonbuild.com/Reference-manual.html#compiler-object.)


I think this is too wide. We *do* want the warning in general when enabled for other code, because the test code does compile fine with -Wmaybe-unitialized, and the warning will catch important bugs for us. Please identify why it was failing in packit, and tailor the exclusion to that specific case. As mentioned above, I suspect it is the combination of `-O2` and `-flto`.

Reply:
yeah, packit is using `-O2` and `-flto=auto` - conditionally added if those are in `c_args`, let's see if it works